pkg/apis/core/validation/utils.go

```diff
@@ -195,3 +195,52 @@ func ValidateIPFamilies(ipFamilies []core.IPFamily, fldPath *field.Path) field.E
 
 	return allErrs
 }
+
+func validateDependencyWatchdogSettings(settings *core.SeedSettingDependencyWatchdog, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if settings.Endpoint != nil && settings.Weeder != nil {
```

Comment:
Why do we need this? See above:

> You can rather make `gardener-apiserver` always keep `weeder/endpoint` and `probe/prober` in sync. Then all clients can just make use of the new fields and don't need to care about the deprecated old fields anymore.

This will also make this validation simpler.